addons/minio/template/base/install.sh

```diff
@@ -10,10 +10,290 @@ function minio_pre_init() {
+    mkdir "$migration_path"
+    MINIO_MIGRATION_HOSTPATH=$migration_path
+}
+
+# create_fs_migration_deployment creates a minio deployment and make it available through a service called
+# 'minio-migrate-fs-backend'. this new deployment uses the same credentials used by the original minio
+# deployment. if the installation uses host path and not a pvc, a temporary path is requested from the user.
+function create_fs_migration_deployment() {
+    local src="$DIR/addons/minio/__MINIO_DIR_NAME__/migrate-fs"
+    local dst="$DIR/kustomize/minio/migrate-fs"
+
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        ask_user_hostpath_for_migration
+        mkdir -p "$dst/hostpath"
+        render_yaml_file_2 "$src/hostpath/deployment.yaml" > "$dst/hostpath/deployment.yaml"
+        render_yaml_file_2 "$src/hostpath/kustomization.yaml" > "$dst/hostpath/kustomization.yaml"
+        render_yaml_file_2 "$src/hostpath/service.yaml" > "$dst/hostpath/service.yaml"
+        kubectl apply -k "$dst/hostpath/"
+    else
+        mkdir -p "$dst/pvc"
+        render_yaml_file_2 "$src/pvc/deployment.yaml" > "$dst/pvc/deployment.yaml"
+        render_yaml_file_2 "$src/pvc/kustomization.yaml" > "$dst/pvc/kustomization.yaml"
+        render_yaml_file_2 "$src/pvc/service.yaml" > "$dst/pvc/service.yaml"
+        render_yaml_file_2 "$src/pvc/pvc.yaml" > "$dst/pvc/pvc.yaml"
+        kubectl apply -k "$dst/pvc/"
+    fi
+
+    local endpoint
+    endpoint=$(kubectl -n ${MINIO_NAMESPACE} get service minio-migrate-fs-backend | tail -n1 | awk '{ print $3}')
+
+    printf "awaiting minio fs migration readiness\n"
+    if ! spinner_until 120 minio_ready "$endpoint"; then
+        bail "Minio FS Migration API failed to report healthy"
+    fi
+}
+
+# destroy_fs_migration_deployment deletes the temporary minio deployment used to migrate the object
+# storage.
+function destroy_fs_migration_deployment() {
+    local dst="$DIR/kustomize/minio/migrate-fs"
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        kubectl delete -k "$dst/hostpath"
+        return
+    fi
+    kubectl delete -k "$dst/pvc"
+}
+
+# swap_fs_migration_pvs swaps the pv backing the minio deployment with the pv used during the migration.
+function swap_fs_migration_pvs() {
+    local migration_pv=$1
+    local src="$DIR/addons/minio/__MINIO_DIR_NAME__"
+    local dst="$DIR/kustomize/minio"
+
+    kubectl patch pv "$migration_pv" --type=json -p="[{'op': 'remove', 'path': '/spec/claimRef'}]"
+    kubectl delete pvc -n $MINIO_NAMESPACE minio-pv-claim
+
+    MINIO_VOLUME_NAME="$migration_pv"
+    render_yaml_file_2 "$src/migrate-fs/minio-pvc.yaml" > "$dst/migrate-fs/minio-pvc.yaml"
+    kubectl create -f "$dst/migrate-fs/minio-pvc.yaml"
+}
+
+# swap_fs_migration_hostpaths moves the host path used during the migration to the host path used by the
+# original minio pod. a backup is kept under hostpath.bkp directory.
+function swap_fs_migration_hostpaths() {
+    suffix=$(echo $RANDOM | md5sum | head -c 8)
```

Comment:
It looks like there is more of a pattern of the following in the codebase. im not sure what is more correct.

```suggestion
    suffix=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c8)
```